### PR TITLE
Add UploadAttestationOccurrence() method in metadata client interface

### PR DESCRIPTION
**Pull request by ooq**
Add `UploadAttestationOccurrence()` method in metadata client interface and all its implementations.

A helper method `CreateOccurrenceFromAttestation()` and its tests are added in the process to deduplicate code.

The existing `CreateAttestationOccurrence()` will now use `UploadAttestationOccurrence()` to upload attestation after creating it. In the long term, we consider deleting `CreateAttestationOccurrence()` altogether because attestation creation should be the caller's responsibility instead of the metadata client. 

`UploadAttestationOccurrence()` also supports both `GenericSignedAttestation` and `PgpSignedAttestation`, while `CreateAttestationOccurrence()` only supports the latter.

Testing for `UploadAttestationOccurrence()` is covered by existing `CreateAttestationOccurrence()` tests, because `CreateAttestationOccurrence()` is the only caller of `UploadAttestationOccurrence()` as for now.

### CODE REVIEW

File: pkg/kritis/metadata/metadata.go [hunk 1]
```diff
@@ -144,3 +153,59 @@ func GetAttestationsFromOccurrence(occ *grafeas.Occurrence) ([]cryptolib.Attesta
 	}
 	return atts, nil
 }
+
+// CreateOccurrenceFromAttestation creates an occurrence from an attestation by specified signature type.
+// The created occurrence can either be a PgpSignedAttestation occurrence or a GenericSignedAttestation occurrence.
+func CreateOccurrenceFromAttestation(att *cryptolib.Attestation, containerImage string, noteName string, sType SignatureType) (*grafeas.Occurrence, error) {
+	attestation := &attestationpb.Attestation{}
+
```
Comment (nenaddedic) [inline pkg/kritis/metadata/metadata.go:161]: mega-nit: remove newline

File: pkg/kritis/metadata/metadata.go [hunk 2]
```diff
@@ -144,3 +153,59 @@ func GetAttestationsFromOccurrence(occ *grafeas.Occurrence) ([]cryptolib.Attesta
 	}
 	return atts, nil
 }
+
+// CreateOccurrenceFromAttestation creates an occurrence from an attestation by specified signature type.
+// The created occurrence can either be a PgpSignedAttestation occurrence or a GenericSignedAttestation occurrence.
+func CreateOccurrenceFromAttestation(att *cryptolib.Attestation, containerImage string, noteName string, sType SignatureType) (*grafeas.Occurrence, error) {
+	attestation := &attestationpb.Attestation{}
```
Comment (nenaddedic) [inline pkg/kritis/metadata/metadata.go:160]: nil ?

File: pkg/kritis/metadata/metadata.go [hunk 3]
```diff
@@ -144,3 +153,59 @@ func GetAttestationsFromOccurrence(occ *grafeas.Occurrence) ([]cryptolib.Attesta
...
+		attestation = &attestationpb.Attestation{
+			Signature: &attestationpb.Attestation_GenericSignedAttestation{
+				GenericSignedAttestation: genericSignedAttestation,
+			},
+		}
+	default:
+		return nil, fmt.Errorf("unknown signature type %v", sType)
+	}
+
+	attestationDetails := &grafeas.Occurrence_Attestation{
```
Comment (nenaddedic) [inline pkg/kritis/metadata/metadata.go:199]: NIt: Can you inline this below?

File: pkg/kritis/metadata/metadata.go [hunk 4]
```diff
@@ -144,3 +153,59 @@ func GetAttestationsFromOccurrence(occ *grafeas.Occurrence) ([]cryptolib.Attesta
 	}
 	return atts, nil
 }
+
+// CreateOccurrenceFromAttestation creates an occurrence from an attestation by specified signature type.
+// The created occurrence can either be a PgpSignedAttestation occurrence or a GenericSignedAttestation occurrence.
+func CreateOccurrenceFromAttestation(att *cryptolib.Attestation, containerImage string, noteName string, sType SignatureType) (*grafeas.Occurrence, error) {
```
Comment (nenaddedic) [inline pkg/kritis/metadata/metadata.go:159]: Do we want this to be an exported function?

File: pkg/kritis/metadata/metadata.go (hunk 1 above)
Comment (ooq) [reply pkg/kritis/metadata/metadata.go]: Done.

File: pkg/kritis/metadata/metadata.go (hunk 3 above)
Comment (ooq) [reply pkg/kritis/metadata/metadata.go]: Done.

File: pkg/kritis/metadata/metadata.go (hunk 2 above)
Comment (ooq) [reply pkg/kritis/metadata/metadata.go:160]: nil doesn't seem to make compiler happy :(

File: pkg/kritis/metadata/metadata.go (hunk 4 above)
Comment (ooq) [reply pkg/kritis/metadata/metadata.go:159]: It needs to be exported to be used in `containeranalysis.go`, etc.

**Pull request merged**